doc-content/shared-kie-docs/src/main/asciidoc/KieServer/controller-rest-api-instances-ref.adoc

```diff
@@ -0,0 +1,271 @@
+      "sever-instance-id": "default-kieserver-instance@localhost:8080",
+      "container-release-id": {
+        "group-id": "evaluation",
+        "artifact-id": "evaluation",
+        "version": "1.0.0-SNAPSHOT"
+      },
+      "messages": [
+        {
+          "severity": "INFO",
+          "timestamp": {
+            "java.util.Date": 1540840061467
+          },
+          "content": [
+            "Release id successfully updated for container evaluation_1.0.0-SNAPSHOT"
+          ]
+        }
+      ],
+      "status": "STARTED"
+    }
+  ]
+}
+----
+
+== [PUT] /server/{serverInstanceId}
+
+Configures a new {KIE_SERVER} instance with the {CONTROLLER}.
+
+.Request parameters
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`serverInstanceId`
+|ID of the new {KIE_SERVER} instance (example: `new-kieserver-instance`)
+|String
+|Required
+
+|*body*
+|A map containing the `id`, `version`, `name`, `location`, `capabilities` (such as `KieServer`, `BPM`, `DMN`, `Swagger`), and other components of the new {KIE_SERVER} instance
+|Request body
+|Required
+|===
+
+.Example request body (JSON)
+[source,json]
+----
+{
+  "id": "new-kieserver-instance",
+  "name": "new-kieserver-instance",
+  "version": "7.14.0-SNAPSHOT",
+  "location": "http://localhost:8080/kie-server/services/rest/server",
+  "capabilities": [
+    "KieServer",
+    "BRM",
+    "BPM",
+    "CaseMgmt",
+    "BPM-UI",
+    "BRP",
+    "DMN",
+    "Swagger"
+  ],
+  "messages": null
+}
+----
+
+*<@Cristiano: The request works but the server responds with null contents of server configs, containers, etc., regardless of payload. Better to update the server so that it responds with the following, or similar.>*
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "response": [
+    {
+      "type": "SUCCESS",
+      "msg": "Server instance new-kieserver-instance@localhost:8080 successfully created."
+    }
+  ]
+}
+----
+
+== [DELETE] /server/{serverInstanceId}
+
+Removes a {KIE_SERVER} instance from the {CONTROLLER}.
+
+.Request parameters
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`serverInstanceId`
+|ID of the {KIE_SERVER} instance to be removed (example: `default-kieserver-instance@localhost:8080`)
+|String
+|Required
+
+|`location`
+|URL of the {KIE_SERVER} instance to be removed
+|String
+|Required
+|===
+
+*<@Cristiano: Please note especially the `location` parameter, which is not clearly documented in the Swagger doc.>*
+
+.Example DELETE endpoint with parameters
+[source,subs="attributes+"]
+----
+http://localhost:8080/{URL_COMPONENT_CENTRAL}/rest/controller/server/new-kieserver-instance?location=http://localhost:8080/kie-server/services/rest/server
+----
+
+*<@Cristiano: The request works but the server responds "undocumented". Need to update the server so that it responds with the following, or similar.>*
```

Comment:
@sterobin DELETE operations return a _204 No Content_ to indicate a successful deletion. That is consistent in both APIs and I would probably keep as is. Any different response code will indicate a problem with a reason. @mswiderski wdyt?

Reply:
I know that 201/204 responses indicate success, but I'm suggesting that this is an inferior, less user-friendly way for the server to respond. For optimal usability, the server should respond with a message for all requests. Successful operations should respond with a "SUCCESS" message and a summary of what was successfully completed, as I have indicated (e.g., "Server instance <instanceId> successfully created."). And no, this is  actually not consistent across the two APIs. Many endpoints in the KIE Server API respond with such success messages. I've marked all of the endpoints in both APIs that respond only with 201/204 and that should respond as I've indicated to be consistent and user-friendly across the board. I understand if this requires more work than can be done today, but this really should be done at some point, imo.

Reply:
@cristianonicolai Understood. I know that 204 responses must not have a message body. I'm suggesting that DELETE requests can have a 202 response and therefore have a message, instead of a non-message 204: https://restfulapi.net/http-methods/#delete.

Either way, if you feel that DELETE requests should stay 204's without a message nonetheless, I can go with that. But all other non-DELETE requests really shouldn't be 204's and ought to have a message, for clarity and for consistency across the API. 

So are you saying as a compromise, that all DELETE's should be non-message 204s, and all others 200's or 202's with message? I'm okay with that.

Reply:
@sterobin correct, if we change to 200 then we could add the content, my concern with that is that it could be considered a breaking change for clients already expecting a 204. @mswiderski wdyt?